codegen/src/api/mod.rs

```diff
@@ -114,24 +109,20 @@ impl RuntimeGenerator {
     pub fn generate_runtime(
         &self,
         item_mod: syn::ItemMod,
-        derives: DerivesRegistry,
-        type_substitutes: TypeSubstitutes,
+        derives: scale_typegen::DerivesRegistry,
+        type_substitutes: scale_typegen::TypeSubstitutes,
         crate_path: syn::Path,
         should_gen_docs: bool,
```

Comment:
Is there any particular reason for having this? It is only passed to the error generation, while others use the `type_gen` settings

Reply:
What do you mean? Having `scale_typegen::DerivesRegistry` specified fully, is not necassary, I should just import `DerivesRegistry`. Maybe I understand the question wrong.

Reply:
Ops, I forgot to type `should_gen_docs`

Reply:
Ah okay, the `should_gen_docs` we actually need because the typegen settings are not built up yet. They are only built in this very function using the `crate_path`, `should_gen_docs`, substitutes and derives :)